docs/static/security/users.asciidoc

```diff
@@ -1,47 +1,76 @@
+You can use role-based access control to grant users access to secured
+resources. The roles that you set up depend on your organization's security
+requirements and the minimum privileges required to use specific features.
 
-To use Logstash with a secured cluster, you need to configure authentication
-credentials for Logstash. Logstash throws an exception and the processing
-pipeline is halted if authentication fails.
+Typically you need the create the following roles:
 
-If encryption is enabled on the cluster, you also need to enable TLS/SSL in the
-Logstash configuration.
+* <<create-writer-role,writer role>> for publishing events
+* <<create-reader-role,reader role>> so that users can access the indices {ls} creates
 
-If you want to monitor your Logstash instance with {monitoring}, and store the
-monitoring data in a secured {es} cluster, you must configure Logstash
-with a username and password for a user with the appropriate permissions.
+{es-security-features} provide {ref}/built-in-roles.html[built-in roles] that
+grant a subset of the privileges needed by {ls} users. 
+For example, `logstash_admin` is a built-in role that provides access to `.logstash-*`
+indices for managing configurations.
+
+When possible, use the built-in roles to minimize the affect of future changes
+on your security strategy.
+
+NOTE: You can assign roles and privileges to **API keys** as an alternative to
+using usernames and passwords. See <<ls-api-keys>> for more information on that
+alternate approach to security.
 
-In addition to configuring authentication credentials for Logstash, you need
-to grant authorized users permission to access the Logstash indices.
 
-[float]
 [[ls-http-auth-basic]]
-==== Configuring Logstash to use Basic Authentication
+==== Configure Logstash to use basic authentication
+[subs="attributes"]
 
-Logstash needs to be able to manage index templates, create indices,
+Logstash must be able to manage index templates, create indices,
 and write and delete documents in the indices it creates.
 
-To set up authentication credentials for Logstash:
+To set up authentication credentials for Logstash, you need to:
 
-. Use the the **Management > Roles** UI in {kib} or the `role` API to create a
-`logstash_writer` role. For *cluster* privileges, add `manage_index_templates` and `monitor`. 
-For *indices* privileges, add `write`, `create`, `delete`, and `create_index`.
+. Create a <<create-writer-role,writer role>>, and grant it cluster and indices
+privileges.
 +
-If you plan to use {ref}/getting-started-index-lifecycle-management.html[index lifecycle
-management], also add `manage_ilm` for cluster and `manage` and `manage_ilm` for indices.
+Example: `logstash_writer`
+. Create an <<create-internal-user,internal user>>, and assign the
+`logstash_writer` role to it.
 +
+Example: `logstash_internal`
+
+. Configure {ls} to <<auth-int-user,authenticate using the internal user>>.
+
+
+[[create-writer-role]]
+===== Create a writer role
```

Comment:
Do users need to create a new role, or can they use the built-in `logstash_system` user that comes with Elasticsearch? If using the built-in user simplifies the configuration, maybe include a note to say, "Just use this built-in user. If you need a separate role, here's how to create one and the privileges you'll need."